TextToSpeech/impl/TTSSpeaker.cpp

```diff
@@ -350,6 +402,58 @@ int TTSSpeaker::speak(TTSSpeakerClient *client, uint32_t id, std::string text, b
     return 0;
 }
 
+void TTSSpeaker::play(string url,SpeechData &data)
+{
+    g_object_set(G_OBJECT(m_source), "location", url.c_str(), NULL);
+    // PCM Sink seems to be accepting volume change before PLAYING state
+    g_object_set(G_OBJECT(m_audioVolume), "volume", (double) (data.client->configuration()->volume() / MAX_VOLUME), NULL);
+    gst_element_set_state(m_pipeline, GST_STATE_PLAYING);
+    #if defined(PLATFORM_AMLOGIC) || defined(PLATFORM_REALTEK)
+    setMixGain(MIXGAIN_PRIM,data.primVolDuck);
+    #endif
+    TTSLOG_VERBOSE("Speaking.... ( %d, \"%s\")", data.id, data.text.c_str());
+    //Wait for EOS with a timeout incase EOS never comes
```

Comment:
Having a new line before & after gst_element_set_state(), before the commit //Wait for EOS would be more readable